packages/playwright/src/fixture.ts

```diff
@@ -255,26 +276,45 @@ export async function replayFixture(
     },
     function handleStepEnd(step) {
       if (expectSteps.has(step.stepId)) {
-        handlePlaywrightEvent("step:end", step.stepId, undefined, {
-          error: step.error ? parseError(step.error) : null,
+        handlePlaywrightEvent({
+          event: "step:end",
+          stepId: step.stepId,
+          params: undefined,
+          detail: {
+            error: step.error ? parseError(step.error) : null,
+          },
         });
       }
     }
   );
 
   const csiListener: ClientInstrumentationListener = {
-    onApiCallBegin: (apiName, params, stackTrace, _wallTime) => {
+    onApiCallBegin: (apiName, params, stackTraceOrFrames, _wallTime) => {
```

Comment:
A utility fn that returns the appropriate `csiListener` for the current version of playwright might this more understandable vs growing the adapter for each internal API change

Reply:
I agree but this is a simple change and I think the code is still OK. It's also not completely apparent how to check the playwright version that is being used by the user. The `playwright` object that gets passed to `replayFixture` doesn't contain this information